psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -308,75 +309,89 @@ def scf_iterate(self, e_conv=None, d_conv=None):
 
         status = []
 
+        # SAD: form initial orbitals from the initial Fock matrix, and
+        # reset the occupations. From here on, the density matrices
+        # are correct.
+        if (self.iteration_ == 0) and self.sad_:
+            self.form_initial_C()
+            self.reset_occupation()
+            self.find_occupation()
+
         # We either do SOSCF or DIIS
-        if (soscf_enabled and (self.iteration_ > 3) and (Drms < core.get_option('SCF', 'SOSCF_START_CONVERGENCE'))):
-
-            Drms = self.compute_orbital_gradient(False, core.get_option('SCF', 'DIIS_MAX_VECS'))
-            diis_performed = False
-            if self.functional().needs_xc():
-                base_name = "SOKS, nmicro="
-            else:
-                base_name = "SOSCF, nmicro="
-
-            if not _converged(Ediff, Drms, e_conv=e_conv, d_conv=d_conv):
-                nmicro = self.soscf_update(
-                    core.get_option('SCF', 'SOSCF_CONV'),
-                    core.get_option('SCF', 'SOSCF_MIN_ITER'),
-                    core.get_option('SCF', 'SOSCF_MAX_ITER'), core.get_option('SCF', 'SOSCF_PRINT'))
-                if nmicro > 0:
-                    # if zero, the soscf call bounced for some reason
-                    self.find_occupation()
-                    status.append(base_name + str(nmicro))
-                    soscf_performed = True  # Stops DIIS
-                else:
-                    if verbose > 0:
-                        core.print_out("Did not take a SOSCF step, using normal convergence methods\n")
-                    soscf_performed = False  # Back to DIIS
+        else:
```

Comment:
I would try to not indent this entire block for readability. You might want to try moving lines 312-319 to 354 as SOSCF cannot happen on the first iteration. This isn't optimal for a number of reasons, but 1) ensures that the correct if/else is paired together so it is easier to see what is happening there and 2) prevents too much indention.

Reply:
Done